app/models/icon.rb

```diff
@@ -10,4 +10,8 @@ class Icon < ApplicationRecord
   has_one :portfolio_item, :dependent => :destroy
 
   validates :image_id, :presence => true
+
+  after_discard do
+    restore_to.update!(:icon_id => nil)
```

Comment:
Does this affect when the discard is initiated at the Portfolio or PortfolioItem level?  Meaning if we discard the PortfolioItem will it lose it's icon?

Reply:
I don't think so, I think it only would do that if the icon was a destroy dependency.

Reply:
No, it won't lose its icon unless we were to add `:icons` here: https://github.com/lindgrenj6/catalog-api/blob/63f8b676d3bf67953c55b3904ace25c9dd796395/app/models/portfolio_item.rb#L5

Reply:
So this brings up an interesting point. From the perspective of lines 9, 10 ( existing behavior before this PR )
```
 has_one :portfolio, :dependent => :destroy
  has_one :portfolio_item, :dependent => :destroy
```
If `destroy` is called on an icon, then the object relation will call up the chain and destroy the associated `Portfolio`, `PortfolioItem` which seems off to me.  

But again, that behavior existed before this PR was here.

Reply:
Ouch, yeah that is definitely not what we want. We don't use `Icon#destroy anywhere luckily, but I'll open a PR to update that to `:nullify` to be safe, just in case we were to use `.destroy` instead of `.discard`.